skops/io/tests/test_persist.py

```diff
@@ -107,11 +108,41 @@ def _tested_estimators(type_filter=None):
     )
 
 
+def _is_steps_like(obj):
+    # helper function to check if an object is something like Pipeline.steps,
+    # i.e. a list of tuples of names and estimators
+    if not isinstance(obj, list):  # must be a list
+        return False
+
+    if not obj:  # must not be empty
+        return False
+
+    if not isinstance(obj[0], tuple):  # must be list of tuples
+        return False
+
+    lens = set(map(len, obj))
+    if not lens == {2}:  # all elements must be length 2 tuples
+        return False
+
+    keys, vals = list(zip(*obj))
+
+    if len(keys) != len(set(keys)):  # keys must be unique
+        return False
+
+    if not all(map(lambda x: isinstance(x, (type(None), BaseEstimator)), vals)):
+        # values must be BaseEstimators or None
+        return False
+
+    return True
+
+
 def _assert_vals_equal(val1, val2):
     if hasattr(val1, "__getstate__"):
         # This includes BaseEstimator since they implement __getstate__ and
         # that returns the parameters as well.
-        _assert_vals_equal(val1.__getstate__(), val2.__getstate__())
+        assert_params_equal(val1.__getstate__(), val2.__getstate__())
```

Comment:
how come removing the `_`?

Reply:
It's a different function vals => params